File: Funciones/11. mayor_igual_tres_numeros/js/mayor.js
/**
 * Mayor igual de 3 números
 * Autor: Santiago GM
 * Fecha: 03/04/2024
 */

let num;
let numUno;
let numDos;
let numTres;

//Como paramétro

function mayorIgual(pnum){
    num = pnum;
    return num;

}

function validar(pnumUno,pnumDos,pnumTres){

    numUno = pnumUno;
    numDos = pnumDos;
    numTres = pnumTres;

    if(numUno == numDos && numUno == numTres &&  numDos == numTres) {
        return "Los tres numeros son iguales";
    }else if (numUno > numDos && numUno > numTres) {
        return `El número uno es mayor que número dos y que número tres: El número uno es ${numUno}, El número dos es ${numDos}, El número tres es ${numTres}`;
    } else if(numDos > numUno && numDos > numTres){
        return `El número dos es mayor que número uno y que número tres: El número uno es ${numUno}, El número dos es ${numDos}, El número tres es ${numTres}`;
    }else{

        return `El número tres es mayor que número uno y que número dos: El número uno es ${numUno}, El número dos es ${numDos}, El número tres es ${numTres}`;
    }
}

//Como Expresión

const mayorIgualExp = function (pnum){
    num = pnum;
    return num;
}
const validarExp =   function(pnumUno,pnumDos,pnumTres){
    numUno = pnumUno;
    numDos = pnumDos;
    numTres = pnumTres;


    if(numUno == numDos && numUno == numTres &&  numDos == numTres) {
        return "Los tres numeros son iguales";
    }else if (numUno > numDos && numUno > numTres) {
        return `El número uno es mayor que número dos y que número tres: El número uno es ${numUno}, El número dos es ${numDos}, El número tres es ${numTres}`;
    } else if(numDos > numUno && numDos > numTres){
        return `El número dos es mayor que número uno y que número tres: El número uno es ${numUno}, El número dos es ${numDos}, El número tres es ${numTres}`;
    }else{

        return `El número tres es mayor que número uno y que número tres: El número uno es ${numUno}, El número dos es ${numDos}, El número tres es ${numTres}`;
    }
}

File: python/11. mayor_igual_tres_numeros/utils/mayor.py
#Mayor igual de 3 números
#Autor: Santiago GM

#Como paramétro

def validar(num_uno_p, num_dos_p, num_tres_p) :
  num_uno, num_dos, num_tres = num_uno_p, num_dos_p, num_tres_p

  if (num_uno == num_dos and num_uno == num_tres and num_dos == num_tres) :
    return "Los tres numeros son iguales"
  elif (num_uno > num_dos and num_uno > num_tres) :
    return f"El número uno es el mayor {num_uno}"

  elif (num_dos > num_uno and num_dos > num_tres) :
    return f"El número dos es el mayor {num_dos}"

  else :
    return f"El número tres es el mayor {num_tres}"


#función lambda
validar_numeros_lamb = lambda num_uno_p, num_dos_p, num_tres_p : "Los tres numeros son iguales" if (num_uno_p == num_dos_p and num_uno_p == num_tres_p and num_dos_p == num_tres_p) else "El número uno es el mayor " + str(num_uno_p) if (num_uno_p > num_dos_p and num_uno_p > num_tres_p) else "El número dos es el mayor " + str(num_dos_p) if (num_dos_p > num_tres_p and num_dos_p > num_uno_p) else "El número tres es el mayor " + str(num_tres_p)